### PR TITLE
Raise `TemplateDoesNotExist` if template does not exist

### LINKED ISSUE #3783: [SES] `send_templated_email` does not raise `TemplateDoesNotExist` with invalid template
How to reproduce:

```python
import boto3
from moto import mock_ses


@mock_ses
def test_ses():
    ses = boto3.client("ses", region_name="us-east-1")
    res = ses.send_templated_email(
        Source="hello@example.com", 
        Destination={"BccAddresses": ["jdoe@example.com"]}, 
        Template="foobar", 
        TemplateData='{}', 
        ConfigurationSetName="blah"
    )
```

Expected:

Template "foobar" does not exist, so the call should raise:

```
botocore.errorfactory.TemplateDoesNotExistException: An error occurred (TemplateDoesNotExist) when calling the SendTemplatedEmail operation: Template foobar does not exist.
```

AWS Docs: https://docs.aws.amazon.com/ses/latest/APIReference/API_SendTemplatedEmail.html

### PR DESCRIPTION
**Pull request by aalvrz**
When sending templated emails

Closes #3783

### CODE REVIEW

File: tests/test_ses/test_ses_boto3.py [hunk 1]
```diff
@@ -132,6 +132,21 @@ def test_send_templated_email():
     conn.send_templated_email.when.called_with(**kwargs).should.throw(ClientError)
 
     conn.verify_domain_identity(Domain="example.com")
+
+    with pytest.raises(ClientError) as ex:
+        conn.send_templated_email(**kwargs).when.called_with(**kwargs)
```
Comment (bblommers) [inline tests/test_ses/test_ses_boto3.py:137]: The `.when.called_with(**kwargs)` part is not necessary here

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @aalvrz, thanks for your PR! Just got one minor niggle, happy to merge once that's fixed

File: tests/test_ses/test_ses_boto3.py (hunk 1 above)
Comment (aalvrz) [reply tests/test_ses/test_ses_boto3.py]: @bblommers Oops. Fixed 👍🏼

**Review by bblommers — APPROVED** [review]
LGTM

**Comment by bblommers** [comment]
This is now part of moto >= 2.0.3.dev7

**Pull request merged**